### ISSUE TITLE
Service level status not reflected in docs for status-{get,set}

### ISSUE THREAD

**Issue by sparkiegeek** [juju/docs#1299]
- https://jujucharms.com/docs/devel/authors-hook-environment#status-set
- https://jujucharms.com/docs/devel/authors-hook-environment#status-get

Both `status-set` and `status-get` hook commands have a `--service` flag which allows the leader of the service to set/get the overall status of the service. This should be documented beyond just the `--help` of the hook tools (hint: `juju run --unit ubuntu/0 'status-get --help'`).

The option has been there since the introduction of the commands, so docs for 1.24 onwards should be updated appopriately.

**Comment by matthewhelmke** [comment]
To help, I did a fresh install/bootstrap/deploy and got the help text referred to above.

`ubuntu@20160812-beta15:~$ juju deploy ubuntu
Added charm "cs:ubuntu-0" to the model.
Deploying charm "cs:ubuntu-0" with the user specified series "xenial".
ubuntu@20160812-beta15:~$ juju status
MODEL    CONTROLLER  CLOUD/REGION         VERSION
default  lxd-test    localhost/localhost  2.0-beta15

APP     VERSION  STATUS   EXPOSED  ORIGIN      CHARM   REV  OS
ubuntu           unknown  false    jujucharms  ubuntu  0    ubuntu

UNIT      WORKLOAD  AGENT  MACHINE  PUBLIC-ADDRESS  PORTS  MESSAGE
ubuntu/0  unknown   idle   0        10.27.236.78           

MACHINE  STATE    DNS           INS-ID         SERIES  AZ
0        started  10.27.236.78  juju-028a1a-0  xenial  



ubuntu@20160812-beta15:~$ juju run --unit ubuntu/0 'status-get --help'
Usage: status-get [options] [--include-data] [--application]

Summary:
print status information

Options:
--application  (= false)
    print status for all units of this application if this unit is the leader
--format  (= smart)
    Specify output format (json|smart|yaml)
--include-data  (= false)
    print all status data
-o, --output (= "")
    Specify an output file

Details:
By default, only the status value is printed.
If the --include-data flag is passed, the associated data are printed also.


ubuntu@20160812-beta15:~$ juju run --unit ubuntu/0 'status-set --help'
Usage: status-set [options] <maintenance | blocked | waiting | active> [message]

Summary:
set status information

Options:
--service, --application  (= false)
    set this status for the application to which the unit belongs if the unit is the leader

Details:
Sets the workload status of the charm. Message is optional.
The "last updated" attribute of the status is set, even if the
status and message are the same as what's already set.`

**Comment by matthewhelmke** [comment]
The help text I included above is from beta15. I haven't been able to check for changes in beta16 due to https://bugs.launchpad.net/juju/+bug/1618636

When I am able to check a newer release, I'll see whether the --service flag has disappeared. Keeping this on hold meanwhile.

**Comment by matthewhelmke** [comment]
Got beta16 working. No change in the help text. Still not enough clarity to move ahead with merging.

```
ubuntu@20160826-beta16:~$ juju run --unit ubuntu/0 'status-get --help'
Usage: status-get [options] [--include-data] [--application]

Summary:
print status information

Options:
--application  (= false)
    print status for all units of this application if this unit is the leader
--format  (= smart)
    Specify output format (json|smart|yaml)
--include-data  (= false)
    print all status data
-o, --output (= "")
    Specify an output file

Details:
By default, only the status value is printed.
If the --include-data flag is passed, the associated data are printed also.
ubuntu@20160826-beta16:~$ juju run --unit ubuntu/0 'status-set --help'
Usage: status-set [options] <maintenance | blocked | waiting | active> [message]

Summary:
set status information

Options:
--service, --application  (= false)
    set this status for the application to which the unit belongs if the unit is the leader

Details:
Sets the workload status of the charm. Message is optional.
The "last updated" attribute of the status is set, even if the
status and message are the same as what's already set.
```

**Comment by matthewhelmke** [comment]
No change in beta17. Will keep watching.

**Comment by matthewhelmke** [comment]
beta18 still includes the help text with both --application and --service.

**Comment by matthewhelmke** [comment]
rc1 still includes both flags.

**Comment by dpb1** [comment]
@matthewhelmke I think you have misunderstood? The OP is asking for the online docs to document the use of --service and --application.  The command --help documentation is fine, but the online docs are out of date.

**Comment by matthewhelmke** [comment]
Thanks, @dpb1 . PR #1300 was written over a month ago to correct the docs. The problem is that during review it was questioned whether --service was going away and we have not found a definitive answer. Checking the help text is as good as I can do without someone responding to questions. As soon as we know with some certainty, it will be easy to either merge or correct that PR and close this out.

**Comment by mitechie** [comment]
Per #1300 all user facing content needs to be application and not service. Anything to the contrary is a bug that needs addressing.

**Comment by matthewhelmke** [comment]
@mitechie thank you. I'll get this PR taken care of and file a bug soon afterward.

**Comment by sparkiegeek** [comment]
Well having application instead of service is all fine and well for 2.x, but this feature was there from the very start of `status-{set,get}` and so I humbly request that 1.x docs are updated too.

**Comment by matthewhelmke** [comment]
@sparkiegeek we can look into that.

@mitechie bug filed - https://bugs.launchpad.net/juju-core/+bug/1626110

**Comment by evilnick** [comment]
Now also fixed for 1.25
